### PR TITLE
Issue #14715: Enforce new naming convention on RecordTypeParameterName in IT area

### PR DESCRIPTION
**Pull request by Lmh-java**
Part of #14715

### CODE REVIEW

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ protected String getCheckName() {
     @Test
     public void testOne() throws Exception {
         final File fileToProcess = new File(getNonCompilablePath(
-                "SuppressionXpathRegressionRecordTypeParameterName1.java"));
+                "InputXpathRecordTypeParameterNameExtend.java"));
```
Comment (MANISH-K-07) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java:44]:
```suggestion
                "InputXpathRecordTypeParameterNameTypeDeclared.java"));
```

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public void testOne() throws Exception {
     @Test
     public void testTwo() throws Exception {
         final File fileToProcess = new File(getNonCompilablePath(
-                "SuppressionXpathRegressionRecordTypeParameterName2.java"));
+                "InputXpathRecordTypeParameterNameDefault.java"));
```
Comment (MANISH-K-07) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java:70]:
```suggestion
                "InputXpathRecordTypeParameterNameTypeDefault.java"));
```

**Review by MANISH-K-07 — COMMENTED** [review]
IMO, it would be better if the record names were same as filenames just like we do with classes by convention.

Items :

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java (hunk 1 above)
Comment (Lmh-java) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java:44]: I don't see the point why you would name it like that. Could you explain it a bit? Basically, why would you add "TypeXXXX" to the file name?

Comment (Lmh-java) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java:44]: The test is testing the generic var with `extends` and `&` synatx.

Comment (MANISH-K-07) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java:44]: Please read about TypeVariables in java.
https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/lang/reflect/TypeVariable.html

Make changes if you feel my suggestions make more sense than suffixes "Extend" and "Default"

Comment (Lmh-java) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java]: Sorry for the late reply, done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java (hunk 2 above)
Comment (Lmh-java) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRecordTypeParameterNameTest.java]: done

**Comment by rnveach** [comment]
@MANISH-K-07 I assume you are done?

**Review by MANISH-K-07 — COMMENTED** [review]
Yes @rnveach , 
I think everything looks good here but I'm not in the power to approve so please approve if you agree with the suggestions..

**Review by rnveach — APPROVED** [review]

**Pull request merged**